### ISSUE TITLE
can not connect to new master redis when old master down

### ISSUE THREAD

**Issue by elwingao** [StackExchange/StackExchange.Redis#2323]
When my master redis is down in my redis sentinel, my dotnet application  will enter ConfigurationChanged event, and new master redis is work. 
But my redis client in my my dotnet application still connect to old master redis in linux server instead of new master redis. 

1 windows server, kill redis master, my redis client will connect to new master redis.
2 windows server, power off old master redis server, my redis client will connect to new master redis.
3  linux server, kill redis master, my redis client will connect to new master redis.
4 linux server, power off old master redis server, my redis client can not connect to new master redis(it still connect to old master redis).

enviroment:
-dotnet core 3.1, centos7.9, redis(have tried serveral version)
-SE.redis (have tried serveral version)

**Comment by elwingao** [comment]
> How are you connecting to the server? Can you should your connection code? There are a few ways to connect to Sentinel, but a generic connect is the only automatic-switching way the library handles (all others are explicit and some are point-in-time), so would need to see your connection logic to advise.

---
"connectStr": "192.168.0.1:26379,192.168.0.2:26379,192.168.0.3:26379,defaultDatabase=0,connectTimeout=5000,connectRetry=2,syncTimeout=5000,password=xxxxxx,serviceName=master"
"serviceName":"master"


-----
ConnectionMultiplexer _conn;
var options = ParseConfigurationOptions(connectStr, ref serviceName);
options.AbortOnConnectFail = false;
options.AllowAdmin = true;
_conn = ConnectionMultiplexer.Connect(options);
                   
            if (options.CommandMap == CommandMap.Sentinel)
            {
                // Get a connection to the cluster's sentinel
                ConfigurationOptions redisServiceOptions = new ConfigurationOptions();
                redisServiceOptions.KeepAlive = 180;
                redisServiceOptions.AllowAdmin = true;
                redisServiceOptions.AbortOnConnectFail = true;
                redisServiceOptions.ServiceName = serviceName;  
                redisServiceOptions.Password = options.Password;                           
                try
                {
                    _conn = _conn.GetSentinelMasterConnection(redisServiceOptions);
                }

**Comment by NickCraver** [comment]
@elwingao You shouldn't need the second section there, by setting a service name on the original options (I can't tell if that's happening here for sure because `ParseConfigurationOptions` isn't ours), and _not_ setting a CommandMap, it'll connect to the Sentinel(s), keep a reference to them and listen for changes to who's the primary. The `.Connect()` call itself will return a connection to the current primary as you want. If that changes, the endpoints will be updated and the connection will reconnect.

In short, this can be as simple as: 
```cs
var primaryConn = ConnectionMultiplexer.Connect("mySentinelServer,serviceName=myService");
```
You can find this in the config docs here: https://stackexchange.github.io/StackExchange.Redis/Configuration
